pkg/db/tx.go

```diff
@@ -34,3 +34,39 @@ func RunInTx(
 	done = true
 	return tx.Commit()
 }
+
```

Comment:
nit. I would propose a simplified function:
```
func RunInTxWithResult[T any](
	ctx context.Context,
	db *sql.DB,
	opts *sql.TxOptions,
	fn func(ctx context.Context, txQueries *queries.Queries) (T, error),
) (result T, err error) {
	tx, err := db.BeginTx(ctx, opts)
	if err != nil {
		return result, err
	}

	// Ensure rollback if not committed
	defer func() {
		if err != nil {
			_ = tx.Rollback()
		}
	}()

	result, err = fn(ctx, queries.New(db).WithTx(tx))
	if err != nil {
		return result, err
	}

	if err = tx.Commit(); err != nil {
		return result, err
	}

	return result, nil
}
```

Reply:
I like that. Updated